# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

Iamm choosing Ranking Signal Analysis. My question is which safe content and search signals are associated with visibility, clicks, engagement, or movement in the starter dataset. I want to understand what actually correlates with performance before committing to any prediction target,this keeps my early work honest and lets me build a stronger baseline or model later, once I know which signals carry real information

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Decision: Which observable content and search signals should a content strategist pay attention to when deciding where to focus review effort before any individual page has been scored or ranked?
Action: A content strategist or editor uses this signal report to decide which types of pages (by position tier, freshness, word count, engagement level) are worth investigating first, guiding where later scoring or modeling work should focus.
Cost of a wrong call: If I highlight a weak or coincidental correlation as meaningful, the team wastes review time chasing a signal that does not actually predict anything real and worse, a genuinely useful signal might get overlooked because my analysis pointed attention elsewhere

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [12]:
!git clone https://github.com/HasnainRaza16/flyrank-ml-hasnain.git

fatal: destination path 'flyrank-ml-hasnain' already exists and is not an empty directory.


In [13]:
import pandas as pd

df = pd.read_csv("flyrank-ml-hasnain/data/raw/content_refresh_anonymized.csv")

print("Total rows:", len(df))
print("Share declining (trend_direction == 'down'):", (df["trend_direction"] == "down").mean())
print("Avg CTR by decline status:")
print(df.groupby(df["trend_direction"] == "down")["ctr"].mean())

Total rows: 30000
Share declining (trend_direction == 'down'): 0.5420666666666667
Avg CTR by decline status:
trend_direction
False    0.731611
True     0.324138
Name: ctr, dtype: float64


Out of 30,000 content items in the starter dataset, 54.2% are currently flagged as declining (trend_direction == "down"). Declining content shows a noticeably lower average CTR (0.32%) compared to non-declining content (0.73%)  less than half. This gap suggests CTR could be one of several signals worth investigating further, though at this stage it's just an observed association, not a confirmed cause of decline. With over half the dataset already declining, this lane looks worth pursuing for the next 7 weeks .There's clearly enough signal and volume to dig into.


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This analysis can say what signals are observed to correlate with content decline in this dataset..for example, declining content shows a lower average CTR than non-declining content. It can offer directional guidance, such as pointing toward CTR as a signal worth investigating further, and it can provide decision-support for a content strategist deciding where to focus review effort.
It cannot claim that low CTR causes decline, or that fixing CTR will reverse a decline that would require an experiment, not just an observed correlation. It also cannot claim anything about Google's actual ranking algorithm, since these are FlyRank's own observable measurements, not insight into search engine internals. Any pattern found here is a reason to look closer, not proof of anything

## Self-check

Before you submit, confirm each line honestly:

- [✓ ] Every section above is filled — markdown thinking AND the code that backs it
- [✓ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [✓] No client names, URLs, or private queries anywhere
- [✓ ] My claims use careful words: observed, measured, directional, decision-support
- [✓ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.